### PR TITLE
<main> Convert test_wl_upgrade to go

### PR DESCRIPTION
**Pull request by lscalabrini01**
Backports:

v2.9 https://github.com/rancher/rancher/pull/46911
v2.8 https://github.com/rancher/rancher/pull/46910
v2.7 https://github.com/rancher/rancher/pull/46875

### CODE REVIEW

**Review by floatingman — DISMISSED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -63,7 +63,7 @@ replace (
 require (
 	github.com/antihax/optional v1.0.0
 	github.com/rancher/rancher/pkg/apis v0.0.0-20240719121207-baeda6b89fe3
-	github.com/rancher/shepherd v0.0.0-20240829194625-ffbf7228bbc0
+	github.com/rancher/shepherd v0.0.0-20240903171436-cf91eb262992
```
Comment (slickwarren) [inline go.mod:66]: this should be updated with a rebase

**Review by slickwarren — COMMENTED** [review]
i think this needs a rebase, otherwise lgtm

**Comment by lscalabrini01** [comment]
> i think this needs a rebase, otherwise lgtm

Yes, I'll rebase again

**Review by floatingman — APPROVED** [review]
LGTM

**Review by slickwarren — DISMISSED** [review]

File: tests/v2/actions/workloads/deployment/deployment.go [hunk 2]
```diff
@@ -99,39 +100,35 @@ func UpdateDeployment(client *rancher.Client, clusterID, namespaceName string, d
...
 }
 
-// GetRolloutHistoryDeployment is a helper to get rollout history deployment
-func GetRolloutHistoryDeployment(client *rancher.Client, clusterID, namespaceName string, deployment *appv1.Deployment) ([]string, error) {
-	deploymentCmd := fmt.Sprintf("deployment.apps/%s", deployment.Name)
-	execCmd := []string{"kubectl", "rollout", "history", "-n", namespaceName, deploymentCmd}
-	logCmd, err := kubectl.Command(client, nil, clusterID, execCmd, "")
-
+// CheckRolloutHistoryDeployment is a helper to get rollout history deployment
+func CheckRolloutHistoryDeployment(client *rancher.Client, clusterID, namespaceName string, deploymentName string, expectedRevision string) error {
```
Comment (slickwarren) [inline tests/v2/actions/workloads/deployment/deployment.go:112]: one last thing 😅 can you just name this to `ValidateRolloutHistoryDeployment`

Comment (lscalabrini01) [reply tests/v2/actions/workloads/deployment/deployment.go:112]: I can

**Review by slickwarren — APPROVED** [review]
lgtm

**Review by floatingman — APPROVED** [review]
LGTM

**Pull request merged**